superbench/benchmarks/micro_benchmarks/memory_bw_performance.py

```diff
@@ -0,0 +1,148 @@
+                logger.error(
+                    'Unsupported mem_type of bandwidth test - benchmark: {}, mem_type: {}, expected: {}.'.format(
+                        self._name, mem_type, ' '.join(self.__mem_types)
+                    )
+                )
+                return False
+            else:
+                command = os.path.join(self._args.bin_dir, self._bin_name)
+                command += ' --' + mem_type
+                if self._args.mode:
+                    if self._args.mode in self.__mode:
+                        command += ' mode=' + self._args.mode
+                    else:
+                        self._result.set_return_code(ReturnCode.INVALID_ARGUMENT)
+                        logger.error(
+                            'Unsupported mode argument of bandwidth test - benchmark: {}, mode: {}, expected: {}.'.
+                            format(self._name, self._args.mode, ' '.join(self.__mode))
+                        )
+                        return False
+                if self._args.memory:
+                    if self._args.memory in self.__memory:
+                        command += ' memory=' + self._args.memory
+                    else:
+                        self._result.set_return_code(ReturnCode.INVALID_ARGUMENT)
+                        logger.error(
+                            'Unsupported memory argument of bandwidth test - benchmark: {}, memory: {}, expected: {}.'.
+                            format(self._name, self._args.memory, ' '.join(self.__memory))
+                        )
+                        return False
+                self._commands.append(command)
+
+        return True
+
+    def _process_raw_result(self, cmd_idx, raw_output):
+        """Function to parse raw results and save the summarized results.
+
+          self._result.add_raw_data() and self._result.add_result() need to be called to save the results.
+
+        Args:
+            cmd_idx (int): the index of command corresponding with the raw_output.
+            raw_output (str): raw output string of the micro-benchmark.
+
+        Return:
+            True if the raw output string is valid and result can be extracted.
+        """
+        self._result.add_raw_data('raw_output_' + self._args.mem_type[cmd_idx], raw_output)
+
+        mem_bw = -1
+        metric = ''
+        valid = True
+        content = raw_output.splitlines()
+        try:
+            for index, line in enumerate(content):
+                if 'Host to Device Bandwidth' in line:
+                    metric = 'H2D_Mem_BW'
+                elif 'Device to Host Bandwidth' in line:
+                    metric = 'D2H_Mem_BW'
+                elif 'Device to Device Bandwidth' in line:
+                    metric = 'D2D_Mem_BW'
+                elif 'Transfer Size' in line and 'Bandwidth' in line and metric != '':
+                    values = list(filter(None, content[index + 1].split()))
+                    mem_bw = float(values[1])
```

Comment:
will this parsing logic work in other mode?
maybe you can use `bandwidthTest --csv` and parse csv output instead

Reply:
thanks for the suggestions, added '--csv' and use another style to extract the value, please check it